include/seqan/find/find_myers_ukkonen.h

```diff
@@ -856,7 +856,7 @@ finline void
 _myersPreInit(PatternState_<TNeedle, TSpec> &state, False)
 {
     typedef typename Value<TNeedle>::Type TValue;
-    memset(state.bitMasks, 0, (ValueSize<TValue>::VALUE + 1) * sizeof(state.bitMasks[0]));
+    memset(state.bitMasks, 0, ValueSize<TValue>::VALUE * sizeof(state.bitMasks[0]));
```

Comment:
It should be `|sigma| + 1`, because other states also set it to this value:

https://github.com/seqan/seqan/blob/30667bc518fa58673d832b60ffb750f57599cb4c/include/seqan/find/find_myers_ukkonen.h#L546-L549

Reply:
If this triggers a warning again, we have to look at it again.

Reply:
It does. It's accessing 768 in an array of size 256 sometimes. So removing the 1 fixed it, but I didn't have an extensive look at this. Anyway, if it was wrong, I'd hope that some tests fail...

Reply:
It seems that we want something like this:

```suggestion
    memset(state.bitMasks, 0, (ValueSize<TValue>::VALUE + 1) * blockCount);
```

(the sizeof seems to be wrong)

Reply:
Oh man I love seqan2:

https://github.com/seqan/seqan/blob/f5f658343c366c9c3d44ba358ffc9317e78a09ed/include/seqan/find/find_myers_ukkonen.h#L216

So the array has actually that size. So it is correct to remove the (+1).

Reply:
So the question is if `|sigma| * w >= (|sigma| + 1) * blockCount =  |sigma| * blockCount + blockCount`

=>

`w >= blockCount + blockCount / |sigma|`

and blockCount is defined as

`block_count = ceiling ( (|query| - 1) / w );`

that means block_count is increasing in query size.

That means a static array is really strange.

Reply:
So it should be replaced at the other places as well?

Reply:
Okay I love seqan even more:

https://github.com/seqan/seqan/blob/f5f658343c366c9c3d44ba358ffc9317e78a09ed/include/seqan/find/find_myers_ukkonen.h#L360

I remember how bad this module was, the fix as it is seems to be a bit more correct. :)

Reply:
> So it should be replaced at the other places as well?

No, the fix is right as it is.

Reply:
:thinking: We might also want to change the static array size to (+1) to match the initial statement.

This might be safer overall.